compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECreateAIEWorkgroup.cpp

```diff
@@ -114,41 +114,68 @@ LogicalResult WorkgroupBuilder::buildForDmaCpyNdOp(
+             << "`amdaie.logicalobjectfifo.from_memref` expected as source";
+    }
+    auto type = cast<LogicalObjectFifoType>(dmaOp.getSource().getType());
+    auto placeholder =
+        rewriter.createAndLookup<AMDAIE::LogicalObjectFifoPlaceholderOp>(
+            rewriter.getUnknownLoc(), type, logicalObjFifo.getTiles());
+    circularDmaSource = placeholder.getResult();
+    circularDmaTarget = dmaOp.getTarget();
     circularDmaTargetOffsets = npuDmaTargetOffsets;
     circularDmaTargetSizes = npuDmaTargetSizes;
     circularDmaTargetStrides = npuDmaTargetStrides;
 
+    npuDmaSource = dmaOp.getSource();
     npuDmaTargetOffsets = empty;
     npuDmaTargetSizes = empty;
     npuDmaTargetStrides = empty;
   } else if (!targetMemSpace) {
     // Check if the target of DmaCpyNd op is from L3 - then target addressing
     // will be controlled by the uController and source addressing will stay in
     // the circular DMA to be part of the AIE configuration.
+    auto logicalObjFifo = dyn_cast<AMDAIE::LogicalObjectFifoFromMemrefOp>(
+        dmaOp.getTarget().getDefiningOp());
+    if (!logicalObjFifo) {
+      return dmaOp.emitOpError()
+             << "`amdaie.logicalobjectfifo.from_memref` expected as source";
+    }
+    auto type = cast<LogicalObjectFifoType>(dmaOp.getTarget().getType());
+    auto placeholder =
+        rewriter.createAndLookup<AMDAIE::LogicalObjectFifoPlaceholderOp>(
+            rewriter.getUnknownLoc(), type, logicalObjFifo.getTiles());
+    circularDmaSource = dmaOp.getSource();
+    circularDmaTarget = placeholder.getResult();
     circularDmaSourceOffsets = npuDmaSourceOffsets;
     circularDmaSourceSizes = npuDmaSourceSizes;
     circularDmaSourceStrides = npuDmaSourceStrides;
 
+    npuDmaTarget = dmaOp.getTarget();
     npuDmaSourceOffsets = empty;
     npuDmaSourceSizes = empty;
     npuDmaSourceStrides = empty;
   }
   auto newDmaOp = rewriter.createAndMap<AMDAIE::CircularDmaCpyNdOp>(
-      rewriter.getUnknownLoc(), dmaOp, dmaOp.getTarget(),
+      rewriter.getUnknownLoc(), dmaOp, circularDmaTarget,
       circularDmaTargetOffsets, circularDmaTargetSizes,
-      circularDmaTargetStrides, dmaOp.getSource(), circularDmaSourceOffsets,
+      circularDmaTargetStrides, circularDmaSource, circularDmaSourceOffsets,
       circularDmaSourceSizes, circularDmaSourceStrides);
 
   IRRewriter::InsertPoint dmaInsertionPoint = rewriter.saveInsertionPoint();
   controlCodeRewriter.setInsertionPoint(controlCode, controlCodeEnd);
   auto npuDmaCpy = controlCodeRewriter.createAndLookup<AMDAIE::NpuDmaCpyNdOp>(
-      loc, newDmaOp.getResult(), npuDmaTargetOffsets, npuDmaTargetSizes,
-      npuDmaTargetStrides, npuDmaSourceOffsets, npuDmaSourceSizes,
-      npuDmaSourceStrides, nullptr, nullptr);
+      loc, newDmaOp.getResult(), npuDmaTarget, npuDmaTargetOffsets,
+      npuDmaTargetSizes, npuDmaTargetStrides, nullptr, npuDmaSource,
+      npuDmaSourceOffsets, npuDmaSourceSizes, npuDmaSourceStrides, nullptr);
```

Comment:
Just add an argument comment for the nullptr value.

Eg: `/*bd_id=*/nullptr`

Reply:
Thanks, done!